src/wallet/wallet.cpp

```diff
@@ -1768,6 +1768,13 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
             // scan succeeded, record block as most recent successfully scanned
             result.last_scanned_block = block_hash;
             result.last_scanned_height = block_height;
+
+            if (save_progress) {
+                CBlockLocator loc= m_chain->getLocator(block_hash);
+
+                WalletBatch batch(GetDatabase());
+                batch.WriteBestBlock(loc);
```

Comment:
I initially considered that the wallet scans tend to be much faster than the indices (unless the wallet is old and hasn't been used for a while), so maybe it would be better to save every block.
I can change this if necessary.

Reply:
Done in https://github.com/bitcoin/bitcoin/commit/37f192b435b03c742aa5f065d1d2d537c4002c94. Thanks.